### PR TITLE
HBASE-26010 Backport HBASE-25703 and HBASE-26002 to branch-2.3

### PR DESCRIPTION
**Pull request by brfrn169**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 11s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ branch-2.3 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 29s |  branch-2.3 passed  |
| +1 :green_heart: |  compile  |   6m  6s |  branch-2.3 passed  |
| +1 :green_heart: |  checkstyle  |   2m 10s |  branch-2.3 passed  |
| +1 :green_heart: |  spotbugs  |   7m  6s |  branch-2.3 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 13s |  the patch passed  |
| +1 :green_heart: |  compile  |   6m  2s |  the patch passed  |
| +1 :green_heart: |  cc  |   6m  2s |  the patch passed  |
| +1 :green_heart: |  javac  |   6m  2s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 41s |  hbase-client: The patch generated 1 new + 375 unchanged - 1 fixed = 376 total (was 376)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  17m 11s |  Patch does not cause any errors with Hadoop 2.10.0 or 3.1.2 3.2.1.  |
| +1 :green_heart: |  hbaseprotoc  |   2m 24s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   7m 49s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 50s |  The patch does not generate ASF License warnings.  |
|  |   |  68m 31s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3396/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3396 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti checkstyle compile cc hbaseprotoc prototool |
| uname | Linux 560b7b0d8552 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2.3 / 6147c1cfae |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3396/1/artifact/yetus-general-check/output/diff-checkstyle-hbase-client.txt |
| Max. process+thread count | 96 (vs. ulimit of 12500) |
| modules | C: hbase-protocol-shaded hbase-protocol hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3396/1/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 31s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  8s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2.3 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 11s |  branch-2.3 passed  |
| +1 :green_heart: |  compile  |   2m 57s |  branch-2.3 passed  |
| +1 :green_heart: |  shadedjars  |   5m 54s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 37s |  branch-2.3 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 22s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m  0s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 54s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 54s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 49s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 35s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 53s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 32s |  hbase-protocol in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 24s |  hbase-client in the patch passed.  |
| -1 :x: |  unit  | 134m 29s |  hbase-server in the patch failed.  |
|  |   | 171m 30s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3396/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3396 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux a5e075d38470 4.15.0-136-generic #140-Ubuntu SMP Thu Jan 28 05:20:47 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2.3 / 6147c1cfae |
| Default Java | AdoptOpenJDK-11.0.10+9 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3396/1/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3396/1/testReport/ |
| Max. process+thread count | 4114 (vs. ulimit of 12500) |
| modules | C: hbase-protocol-shaded hbase-protocol hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3396/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 34s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  9s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2.3 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 38s |  branch-2.3 passed  |
| +1 :green_heart: |  compile  |   2m 29s |  branch-2.3 passed  |
| +1 :green_heart: |  shadedjars  |   5m  6s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 31s |  branch-2.3 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 21s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 29s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 29s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m  6s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 29s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 42s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 27s |  hbase-protocol in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 14s |  hbase-client in the patch passed.  |
| -1 :x: |  unit  | 148m 39s |  hbase-server in the patch failed.  |
|  |   | 181m  1s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3396/1/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3396 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 8f61deda03a3 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2.3 / 6147c1cfae |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3396/1/artifact/yetus-jdk8-hadoop2-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3396/1/testReport/ |
| Max. process+thread count | 3733 (vs. ulimit of 12500) |
| modules | C: hbase-protocol-shaded hbase-protocol hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3396/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 34s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ branch-2.3 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 32s |  branch-2.3 passed  |
| +1 :green_heart: |  compile  |   6m 10s |  branch-2.3 passed  |
| +1 :green_heart: |  checkstyle  |   2m 10s |  branch-2.3 passed  |
| +1 :green_heart: |  spotbugs  |   7m  7s |  branch-2.3 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 11s |  the patch passed  |
| +1 :green_heart: |  compile  |   5m 58s |  the patch passed  |
| +1 :green_heart: |  cc  |   5m 58s |  the patch passed  |
| +1 :green_heart: |  javac  |   5m 58s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 11s |  The patch passed checkstyle in hbase-protocol-shaded  |
| +1 :green_heart: |  checkstyle  |   0m 11s |  The patch passed checkstyle in hbase-protocol  |
| +1 :green_heart: |  checkstyle  |   0m 40s |  hbase-client: The patch generated 0 new + 375 unchanged - 1 fixed = 375 total (was 376)  |
| +1 :green_heart: |  checkstyle  |   1m  4s |  The patch passed checkstyle in hbase-server  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  16m 59s |  Patch does not cause any errors with Hadoop 2.10.0 or 3.1.2 3.2.1.  |
| +1 :green_heart: |  hbaseprotoc  |   2m 25s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   7m 55s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 51s |  The patch does not generate ASF License warnings.  |
|  |   |  67m 57s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3396/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3396 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti checkstyle compile cc hbaseprotoc prototool |
| uname | Linux 33cf662e5239 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2.3 / 6147c1cfae |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| Max. process+thread count | 96 (vs. ulimit of 12500) |
| modules | C: hbase-protocol-shaded hbase-protocol hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3396/2/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 39s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  6s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2.3 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 52s |  branch-2.3 passed  |
| +1 :green_heart: |  compile  |   3m  6s |  branch-2.3 passed  |
| +1 :green_heart: |  shadedjars  |   6m 37s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 46s |  branch-2.3 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 22s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 28s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 21s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 21s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   6m 54s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 44s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 54s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 32s |  hbase-protocol in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 29s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  | 132m 20s |  hbase-server in the patch passed.  |
|  |   | 173m 21s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3396/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3396 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux b6b91a9028d0 4.15.0-136-generic #140-Ubuntu SMP Thu Jan 28 05:20:47 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2.3 / 6147c1cfae |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3396/2/testReport/ |
| Max. process+thread count | 4428 (vs. ulimit of 12500) |
| modules | C: hbase-protocol-shaded hbase-protocol hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3396/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m  3s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  7s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2.3 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 46s |  branch-2.3 passed  |
| +1 :green_heart: |  compile  |   2m 42s |  branch-2.3 passed  |
| +1 :green_heart: |  shadedjars  |   5m 43s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 31s |  branch-2.3 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 57s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 38s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 38s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 58s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 31s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 47s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 28s |  hbase-protocol in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 13s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  | 141m 57s |  hbase-server in the patch passed.  |
|  |   | 178m 51s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3396/2/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3396 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 0f03852afe5c 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2.3 / 6147c1cfae |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3396/2/testReport/ |
| Max. process+thread count | 3827 (vs. ulimit of 12500) |
| modules | C: hbase-protocol-shaded hbase-protocol hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3396/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache9** [comment]
Ping @ndimiduk

**Comment by ndimiduk** [comment]
I prefer that we reopen the original jiras and backport them one-by-one. The reason being that it's easier to tell directly if a JIRA has been applied to a branch. "One JIRA, One Commit" has been our policy for as long as I can remember. It's okay to have them both in the same pull request, since we want both of them or neither of them, but please preserve the commit/JIRA automaticity.

File: hbase-protocol/src/main/protobuf/MultiRowMutation.proto [hunk 1]
```diff
@@ -37,9 +37,11 @@ message MutateRowsRequest {
   optional uint64 nonce_group = 2;
   optional uint64 nonce = 3;
   optional RegionSpecifier region = 4;
+  repeated Condition condition = 5;
 }
 
 message MutateRowsResponse {
+  optional bool processed = 1;
```
Comment (ndimiduk) [inline hbase-protocol/src/main/protobuf/MultiRowMutation.proto:44]: Is this a backward-incompatible breaking change to the wire format? How does a client generated against the old definition handle the presence of any fields?

**Comment by ndimiduk** [comment]
I'm nervous to make non-bug-fix changes on patch releases, especially because, to the best of my knowledge, the `MultiRowMutationEndpoint` is an "undocumented/internal" coprocessor that is used only by the meta table. What's the motivation of this change to branch-2.3?

**Comment by brfrn169** [comment]
@ndimiduk Thank you for taking look at this!

As it's not a bug fix, I will close this PR. Thanks.

**Pull request closed without merging**